### PR TITLE
feat(ApiKey): add CRUDL for API keys

### PR DESCRIPTION
**Pull request by xjacka**
https://github.ibm.com/Incubation/bee-ui/issues/616

### CODE REVIEW

File: src/administration/api-keys.module.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    },
+    async (req) => updateApiKey({ ...req.params, ...req.body })
+  );
+
+  app.get<{ Querystring: ApiKeysListQuery; Params: ApiKeyListParams }>(
+    '/organization/projects/:project_id/api_keys',
+    {
+      schema: {
+        querystring: apiKeysListQuerySchema,
+        response: { [StatusCodes.OK]: apiKeysListResponseSchema },
```
Comment (pilartomas) [inline src/administration/api-keys.module.ts:106]: missing params schema

File: src/administration/api-keys.service.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+export async function listApiKeys({
+  limit,
+  order,
+  order_by,
+  after,
+  before,
+  project_id
+}: ApiKeysListQuery & ApiKeysListParams): Promise<ApiKeysListResponse> {
+  const filter: FilterQuery<ProjectApiKey> = { project: project_id };
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:106]: `.findOrFail` project to validate it inside the org.

File: src/administration/api-keys.service.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+}
+
+export async function updateApiKey({
+  api_key_id,
+  project_id,
+  name
+}: ApiKeyUpdateParams & ApiKeyUpdateBody): Promise<ApiKeyUpdateResponse> {
+  const apiKey = await ORM.em
+    .getRepository(ProjectApiKey)
+    .findOneOrFail({ id: api_key_id, project: project_id });
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:90]: ditto

File: src/administration/api-keys.service.ts [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  return toDto(apiKey, keyValue);
+}
+
+export async function readApiKey({
+  api_key_id,
+  project_id
+}: ApiKeyReadParams): Promise<ApiKeyReadResponse> {
+  const apiKey = await ORM.em
+    .getRepository(ProjectApiKey)
+    .findOneOrFail({ id: api_key_id, project: project_id });
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:79]: ditto

File: src/administration/dtos/api-key-create.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  properties: {
+    name: { type: 'string' }
+  }
+} as const satisfies JSONSchema;
+export type ApiKeyCreateBody = FromSchema<typeof apiKeyCreateBodySchema>;
+
+export const apiKeyCreateParamsSchema = apiKeyListParamsSchema;
+export type ApiKeyCreateParams = FromSchema<typeof apiKeyCreateParamsSchema>;
+
+export const apiKeyCreateResponseSchema = apiKeySchema;
```
Comment (pilartomas) [inline src/administration/dtos/api-key-create.ts:35]: Create endpoint also returns the secret

File: src/administration/api-keys.service.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import { ORM } from '@/database';
+import { APIError, APIErrorCode } from '@/errors/error.entity';
+
+export function toDto(apiKey: Loaded<ProjectApiKey>, sensitiveId?: string): ApiKeyDto {
+  return {
+    object: 'organization.project.api_key',
+    id: apiKey.id,
+    name: apiKey.name,
+    created_at: dayjs(apiKey.createdAt).unix(),
+    sensitive_id: typeof sensitiveId === 'string' ? sensitiveId : undefined
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:43]: I'd change this to `sensitive_key` or  `secret_key` or something like that. I know it comes from OpenAI but the key is not an ID in our case.

File: src/administration/api-keys.service.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  return createPaginatedResponse(cursor, toDto);
+}
+
+export async function deleteApiKey({
+  api_key_id,
+  project_id
+}: ApiKeyDeleteParams): Promise<ApiKeyDeleteResponse> {
+  const apiKey = await ORM.em
+    .getRepository(ProjectApiKey)
+    .findOneOrFail({ id: api_key_id, project: project_id });
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:122]: ditto

**Review by pilartomas — CHANGES_REQUESTED** [review]

File: src/administration/api-keys.service.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+    sensitive_id: typeof sensitiveId === 'string' ? sensitiveId : undefined
+  };
+}
+
+export async function createApiKey({
+  name,
+  project_id
+}: ApiKeyCreateBody & ApiKeyCreateParams): Promise<ApiKeyCreateResponse> {
+  const authorProjectPrincipal = getProjectPrincipal();
+  const project = await ORM.em.getRepository(Project).findOneOrFail({ id: project_id });
```
Comment (pilartomas) [inline src/administration/api-keys.service.ts:52]: Consider capping the number of keys per project.

Comment (xjacka) [reply src/administration/api-keys.service.ts:52]: Hmm, do you think this is needed?

**Pull request merged**